pkg/util/manifests.go

```diff
@@ -168,3 +171,17 @@ func AppendMissingDefaultMCPManifests(currentMCPs []*mcfgv1.MachineConfigPool) [
 
 	return append(finalMCPList, currentMCPs...)
 }
+
+func DeserializeObjectFromData(data []byte, addToSchemeFn ...func(scheme2 *runtime.Scheme) error) (runtime.Object, error) {
```

Comment:
I tend to like to do the `AddToScheme` once around init phase, but we ca improve this later